app/javascript/components/Study.js

```diff
@@ -11,28 +11,22 @@ export function formatDescription(rawDescription, term) {
   return shortenDescription(textDescription, term)
 }
 
-function highlightText(text, termMatches) {
-  const matchedIndices = []
+export function highlightText(text, termMatches) {
+  let matchedIndices = []
   if (termMatches) {
+    matchedIndices=termMatches.map(term => text.indexOf(term))
+  }
+  if (matchedIndices.length>0) {
     termMatches.forEach((term, index) => {
-      let match
       const regex = RegExp(term, 'gi')
-      // Find indices where match occured
-      while ((match = regex.exec(text)) != null) {
-        matchedIndices.push(match.index)
-      }
+      text = text.replace(regex, `<span id='highlight'>${term}</span>`)
```

Comment:
sorry I missed this on my previous review -- this needs to be a class, not an id